packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -509,14 +568,28 @@ const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = isolatedFie
 		const fieldChange = change.fieldChange;
 		if (fieldChange !== undefined) {
 			inverse.fieldChange = { wasEmpty: fieldChange.newContent === undefined };
+			if (fieldChange.newContent?.changes !== undefined) {
+				// The node inserted by change will be the node deleted by inverse
+				// Move the inverted changes to the child change field
+				inverse.childChange = invertChild(fieldChange.newContent.changes, 0);
+			}
+
 			if (!fieldChange.wasEmpty) {
 				assert(revision !== undefined, 0x592 /* Unable to revert to undefined revision */);
-				inverse.fieldChange.newContent = { revert: reviver(revision, 0, 1)[0] };
+				inverse.fieldChange.newContent = { revert: reviver(revision, 0, 1)[0], revision };
+				if (change.childChange !== undefined) {
+					assert(change.deletedBy === undefined, "Unhandled scenario");
+					inverse.fieldChange.newContent.changes = invertChild(change.childChange, 0);
```

Comment:
If I'm following correctly, `change.deletedBy !== undefined` would mean that the input changeset wants to apply some changes to some node that, as of this changeset's input context, no longer exists because it was deleted by revision `change.deletedBy` (and is not being revived by this changeset).

Inverting a changeset C1 presuposes that C1 was applied as is (as opposed to being further rebased then applied) which means inverting its muted effects should also produce some non-effectful changes. It's tempting to think that we can just drop whatever muted changes it had, but we may want the inverse of those changes to be present in the returned changeset so that a retroactive undo being postbased over this change can pick them up.

Retroactive undo is not something we need to worry about right now, so dropping the muted changes seems fine. Here's my suggestion:
```suggestion
					if (change.deletedBy === undefined) {
						inverse.fieldChange.newContent.changes = invertChild(change.childChange, 0);
					} else {
						// We currently drop the muted changes in the inverse.
						// TODO: produce muted inverse changes so that a retroactive undo of revision
						// `change.deletedBy` would be able to pick up and unmute those changes.
					}
```

Reply:
Committed as suggested.